File: com.bosch.fsp.master_of_plow.gen/src/com/bosch/nevonex/fcal/impl/RECEIVE_HYD_INFO_T_R.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IRECEIVE_HYD_INFO_T_R;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class RECEIVE_HYD_INFO_T_R extends EObjectImpl implements IRECEIVE_HYD_INFO_T_R {
	protected float tRZ_HYD_SIG1_V = 0.0f;
	protected float tRZ_HYD_SIG2_V = 0.0f;
	protected int tRZ_HYD_DIAG = 0;
	protected int tRZ_HYD_AUTO = 0;

	protected RECEIVE_HYD_INFO_T_R() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.RECEIVE_HYD_INFO_T_R;
	}

	public float getTRZ_HYD_SIG1_V() {
		return tRZ_HYD_SIG1_V;
	}

	public void setTRZ_HYD_SIG1_V(float newTRZ_HYD_SIG1_V) {
		tRZ_HYD_SIG1_V = newTRZ_HYD_SIG1_V;
	}

	public float getTRZ_HYD_SIG2_V() {
		return tRZ_HYD_SIG2_V;
	}

	public void setTRZ_HYD_SIG2_V(float newTRZ_HYD_SIG2_V) {
		tRZ_HYD_SIG2_V = newTRZ_HYD_SIG2_V;
	}

	public int getTRZ_HYD_DIAG() {
		return tRZ_HYD_DIAG;
	}

	public void setTRZ_HYD_DIAG(int newTRZ_HYD_DIAG) {
		tRZ_HYD_DIAG = newTRZ_HYD_DIAG;
	}

	public int getTRZ_HYD_AUTO() {
		return tRZ_HYD_AUTO;
	}

	public void setTRZ_HYD_AUTO(int newTRZ_HYD_AUTO) {
		tRZ_HYD_AUTO = newTRZ_HYD_AUTO;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG1_V:
			return getTRZ_HYD_SIG1_V();
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG2_V:
			return getTRZ_HYD_SIG2_V();
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_DIAG:
			return getTRZ_HYD_DIAG();
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_AUTO:
			return getTRZ_HYD_AUTO();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG1_V:
			setTRZ_HYD_SIG1_V((Float) newValue);
			return;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG2_V:
			setTRZ_HYD_SIG2_V((Float) newValue);
			return;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_DIAG:
			setTRZ_HYD_DIAG((Integer) newValue);
			return;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_AUTO:
			setTRZ_HYD_AUTO((Integer) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG1_V:
			setTRZ_HYD_SIG1_V(0.0f);
			return;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG2_V:
			setTRZ_HYD_SIG2_V(0.0f);
			return;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_DIAG:
			setTRZ_HYD_DIAG(0);
			return;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_AUTO:
			setTRZ_HYD_AUTO(0);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG1_V:
			return tRZ_HYD_SIG1_V != 0.0f;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_SIG2_V:
			return tRZ_HYD_SIG2_V != 0.0f;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_DIAG:
			return tRZ_HYD_DIAG != 0;
		case FcalPackage.RECEIVE_HYD_INFO_T_R__TRZ_HYD_AUTO:
			return tRZ_HYD_AUTO != 0;
		}
		return super.eIsSet(featureID);
	}
} //RECEIVE_HYD_INFO_T_R

File: master_of_plow_CPP_SDK/src-gen/nevonex/fcal/RECEIVE_HYD_INFO_T_R.cpp
/*
 * nevonex/fcal/RECEIVE_HYD_INFO_T_R.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "RECEIVE_HYD_INFO_T_R.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(RECEIVE_HYD_INFO_T_R.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
RECEIVE_HYD_INFO_T_R::RECEIVE_HYD_INFO_T_R() : m_tRZ_HYD_SIG1_V(0.0f), m_tRZ_HYD_SIG2_V(0.0f), m_tRZ_HYD_DIAG(0), m_tRZ_HYD_AUTO(0)
{

    /*PROTECTED REGION ID(RECEIVE_HYD_INFO_T_R__RECEIVE_HYD_INFO_T_R) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

RECEIVE_HYD_INFO_T_R::~RECEIVE_HYD_INFO_T_R()
{

    /*PROTECTED REGION ID(RECEIVE_HYD_INFO_T_R__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EFloat RECEIVE_HYD_INFO_T_R::getTRZ_HYD_SIG1_V() const
{
    return m_tRZ_HYD_SIG1_V;
}

void RECEIVE_HYD_INFO_T_R::setTRZ_HYD_SIG1_V(::ecore::EFloat _tRZ_HYD_SIG1_V)
{

    ::ecore::EFloat _old_tRZ_HYD_SIG1_V = m_tRZ_HYD_SIG1_V;
    m_tRZ_HYD_SIG1_V = _tRZ_HYD_SIG1_V;

}

::ecore::EFloat RECEIVE_HYD_INFO_T_R::getTRZ_HYD_SIG2_V() const
{
    return m_tRZ_HYD_SIG2_V;
}

void RECEIVE_HYD_INFO_T_R::setTRZ_HYD_SIG2_V(::ecore::EFloat _tRZ_HYD_SIG2_V)
{

    ::ecore::EFloat _old_tRZ_HYD_SIG2_V = m_tRZ_HYD_SIG2_V;
    m_tRZ_HYD_SIG2_V = _tRZ_HYD_SIG2_V;

}

::ecore::EInt RECEIVE_HYD_INFO_T_R::getTRZ_HYD_DIAG() const
{
    return m_tRZ_HYD_DIAG;
}

void RECEIVE_HYD_INFO_T_R::setTRZ_HYD_DIAG(::ecore::EInt _tRZ_HYD_DIAG)
{

    ::ecore::EInt _old_tRZ_HYD_DIAG = m_tRZ_HYD_DIAG;
    m_tRZ_HYD_DIAG = _tRZ_HYD_DIAG;

}

::ecore::EInt RECEIVE_HYD_INFO_T_R::getTRZ_HYD_AUTO() const
{
    return m_tRZ_HYD_AUTO;
}

void RECEIVE_HYD_INFO_T_R::setTRZ_HYD_AUTO(::ecore::EInt _tRZ_HYD_AUTO)
{

    ::ecore::EInt _old_tRZ_HYD_AUTO = m_tRZ_HYD_AUTO;
    m_tRZ_HYD_AUTO = _tRZ_HYD_AUTO;

}

// References

